client/allocrunner/network_hook.go

```diff
@@ -106,7 +108,15 @@ func (h *networkHook) Prerun() error {
 		if err != nil {
 			return fmt.Errorf("failed to configure networking for alloc: %v", err)
 		}
-
+		if hostname, ok := spec.Labels[dockerNetSpecLabelKey]; ok {
+			if len(hostname) > 12 {
+				hostname = hostname[:12]
+			}
```

Comment:
What's so special about 12?

Reply:
The `docker_sandbox_container_id` value is the full container ID, whereas Docker will set the hostname in the pause container as `docker_sandbox_container_id[:12]`. But we should certainly document that here!